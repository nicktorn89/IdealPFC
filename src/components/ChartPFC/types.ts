export interface ChartsProps {
  data: ChartData[];
  size?: number;
}

export type ChartData = {
  name: string;
  value: number;
  color: string;
};
