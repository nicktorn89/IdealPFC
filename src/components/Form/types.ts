import { MainState } from 'src/modules/Main/types';

export interface FormProps {
  values: MainState;
  onChangeField: (e: React.MouseEvent<HTMLInputElement>) => void;
}
