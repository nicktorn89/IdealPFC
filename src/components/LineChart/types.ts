export interface LineChartProps {
  data: LineChartData[];
}

export type LineChartData = {
  name: string;
  calories: number;
};
