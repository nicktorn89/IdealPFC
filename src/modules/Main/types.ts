export interface MainState {
  weight: number;
  height: number;
  age: number;
  desiredWeight?: number;

  [key: string]: number | undefined;
}

export interface MainProps {
}

export interface Nutrients {
  protein: number;
  fats: number;
  carbohydrates: number;
}
