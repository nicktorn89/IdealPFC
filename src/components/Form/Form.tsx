import React, { useState } from 'react';
import { FormProps } from './types';
import { Columns, Section, Form, Button } from 'react-bulma-components/full';

const fields = [
  'height', 'age', 'weight', 'desiredWeight',
];

const labels = [
  'Рост, см', 'Возраст', 'Вес, кг', 'Желаемый вес, кг',
];

const FormComponent: React.FC<FormProps> = ({ values, onChangeField }) => {
  const [isVisible, setVisibility] = useState(false);

  const handleToggle = () => {
    setVisibility(!isVisible);
  };

  const renderFields = fields.map((element, index) => (
    <Columns.Column key={index} >
      <Form.Label>
        {labels[index]}
        <Form.Input 
          name={element}
          value={values[element]}
          onChange={onChangeField}
        />
      </Form.Label>
    </Columns.Column>
  ));

  return (
    <Section>
      <Button onClick={handleToggle}>
        {isVisible ? 'Скрыть данные' : 'Ввести данные'}
      </Button>
      <Columns>
        {isVisible && renderFields}
      </Columns>
    </Section>
  );
};

export default FormComponent;
