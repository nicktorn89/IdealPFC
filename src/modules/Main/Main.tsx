import React from 'react';
import { MainState, MainProps } from './types';
import FormComponent from '../../components/Form';
import HeaderComponent from 'src/components/Header';
import ChartPFC from 'src/components/ChartPFC';
import { PieChartData } from 'src/components/ChartPFC/types';
import LineChart from 'src/components/LineChart';
import { LineChartData } from 'src/components/LineChart/types';

const dataMock: PieChartData[] = [
  {
    name: 'Белки',
    value: 200,
    color: '#FFFF4C',
  },
  {
    name: 'Жиры',
    value: 150,
    color: '#FF4C4C',
  },
  {
    name: 'Углеводы',
    value: 650,
    color: '#4C4CFF',
  },
];

const anotherMock: LineChartData[] = [
  {
    name: '1 неделя',
    calories: 2500,
  },
  {
    name: '2 неделя',
    calories: 2300,
  },
  {
    name: '3 неделя',
    calories: 2100,
  },
  {
    name: '4 неделя',
    calories: 1900,
  },
  {
    name: '5 неделя',
    calories: 1700,
  },
];

export class Main extends React.Component<MainProps, MainState> {
  public readonly state = {
    weight: 0,
    height: 0,
    age: 0,
  };

  public handleFieldChange = ({ target }: React.MouseEvent<HTMLInputElement>) => {
    const { name, value } = target as HTMLInputElement;

    if (isNaN(+value)) return;

    this.setState({ [name]: +value });
  }

  public render = () => {
    return (
      <React.Fragment>
        <HeaderComponent />
        <FormComponent
          values={this.state}
          onChangeField={this.handleFieldChange}
        />
        <ChartPFC data={dataMock} />
        <LineChart data={anotherMock} />
      </React.Fragment>
    );
  }
}

export default Main;
