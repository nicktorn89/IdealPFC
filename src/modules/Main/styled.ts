import styled from 'styled-components';
import { media } from 'src/theme/mixin';
