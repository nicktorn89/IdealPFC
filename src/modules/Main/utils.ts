import { Nutrients } from './types';

export const roundToDecimal = (value: number) => Math.round(value * 10) / 10;

export const countNutrients = (height: number, weight: number, hasSportLoad?: boolean): Nutrients => 
  ({
    protein: roundToDecimal(weight * (hasSportLoad ? 2.5 : 1.5)),
    fats: roundToDecimal(weight * (hasSportLoad ? 1.5 : 0.8)),
    carbohydrates: roundToDecimal(weight * (hasSportLoad ? 4 : 2)),
  });
