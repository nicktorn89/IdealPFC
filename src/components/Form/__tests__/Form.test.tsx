import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { default as Form } from '../Form';
import { MainState } from 'src/modules/Main/types';

describe('<Form />', () => {
  const valuesMock: MainState = { weight: 0, height: 0, age: 0 };
  const voidFunc = () => { return; };

  const FormComponent = shallow(<Form 
    values={valuesMock} 
    onChangeField={voidFunc}
  />);
  const FormComponentToJson = shallowToJson(FormComponent);

  it('should be equal to snapshot', () => {
    expect(FormComponentToJson).toMatchSnapshot();
  });
});
