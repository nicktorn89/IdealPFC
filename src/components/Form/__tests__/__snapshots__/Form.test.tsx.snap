// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should be equal to snapshot 1`] = `
<c
  className=""
  renderAs="section"
  size={null}
  style={Object {}}
>
  <b
    className=""
    color={null}
    disabled={false}
    fullwidth={false}
    inverted={false}
    isSelected={false}
    isStatic={false}
    loading={false}
    onClick={[Function]}
    outlined={false}
    remove={false}
    renderAs="button"
    reset={false}
    rounded={false}
    size={null}
    state={null}
    style={Object {}}
    submit={false}
    text={false}
  >
    Ввести данные
  </b>
  <m
    breakpoint={null}
    centered={false}
    className=""
    gapless={false}
    multiline={true}
    style={Object {}}
  />
</c>
`;
